controllers/provider-alicloud/pkg/alicloud/client/client.go

```diff
@@ -117,6 +117,12 @@ func (c *storageClient) CreateBucketIfNotExists(ctx context.Context, bucketName
 		}
 		return err
 	}
+
+	config := oss.ServerEncryptionRule{SSEDefault: oss.SSEDefaultRule{SSEAlgorithm: "KMS"}}
```

Comment:
If i understood the API correctly, the default CMK generated by OSS will be used for encryption.  What will happen if this key is lost somehow? Will we still be able to decrypt the backups. ALso, one more thing i'm not clear from (doc)[https://www.alibabacloud.com/help/doc-detail/31871.htm?spm=a2c63.p38356.a1.2.380324d3RO5ivi] specially following line
> To encrypt an object when creating it, you can include the x-oss-server-side-encryption header in the request and specify its value to KMS 

 Does setting bucket encryption also encrypts the objects by default. Or for encrypting the object one  has to update the object request compulsory to have above mentioned header.

Reply:
@swapnilgm Thank for your good questions. For your 1st question, if we use KMS to encrypt OSS objects, one default CMK will be generated, and it cannot be deleted when being used, so it should not be lost. For your 2nd question, when the encryption of buckets is enabled, objects uploaded to these buckets will be encrypted by default. I will have a double check tomorrow.

Reply:
>one default CMK will be generated, and it cannot be deleted when being used, so it should not be lost. 

Cool. One more question then, as you mentioned in release notes 
> Buckets already created will not be affected, which means only buckets created after the change will enable SSE.

 On the similar line, will we be able to change the default generated CMK later on to custom CMK? What will happen to old encrypted objects in that case?

> when the encryption of buckets is enabled, objects uploaded to these buckets will be encrypted by default. I will have a double check tomorrow.

Sure. Please confirm and reply. I'll rely on you for this clarification

Reply:
First to correct one my yesterday answer after checked with Alicloud support engineers. Object uploaded will not be encrypted by default, even the bucket has been already enabled SSE. `x-oss-server-side-encryption` header is required. 

> On the similar line, will we be able to change the default generated CMK later on to custom CMK? What will happen to old encrypted objects in that case?

Code on the line is only to enable buckets encrypted using KMS, not directly related to CMK. According to my understanding, custom CMKs usage is on the **objects** level, which means 
we can use different CMKs to encrypt different objects in the same bucket. What need to do is set both both `x-oss-server-side-encryption` and `X-oss-server-side-encrpytion-key-id` in the header.  For exapmle, https://github.com/aliyun/aliyun-oss-go-sdk/blob/master/sample/put_object.go#L72

Reply:
May i know the immediate use case for having server side bucket encryption? Is there any specific requirement on alicloud support?

If not, i'll prefer closing this PR. And take this a task asa common task for all providers.

Reply:
The change is not hurry, so it is fine to close it.